run-tests.php

```diff
@@ -19,29 +19,51 @@
    |          Marcus Boerger <helly@php.net>                              |
    |          Derick Rethans <derick@php.net>                             |
    |          Sander Roobol <sander@php.net>                              |
+   |          Andrea Faulds <ajf@ajf.me>                                  |
    | (based on version by: Stig Bakken <ssb@php.net>)                     |
    | (based on the PHP 3 test framework by Rasmus Lerdorf)                |
    +----------------------------------------------------------------------+
  */
 
 /* $Id$ */
 
-define('INIT_DIR', getcwd());
+run_tests();
+
+function run_tests() {
+    global
+        $argc, $argv,
+        $php, $php_cgi, $phpdbg, $user_tests, $ini_overwrites, $pass_options,
+        $exts_to_test, $valgrind, $no_file_cache, $sum_results,
+        $just_save_results, $PHP_FAILED_TESTS, $output_file, $test_files,
+        $ignored_by_ext, $exts_skipped, $DETAILED, $test_results,
+        $failed_tests_file, $result_tests_file, $test_idx, $cfg, $log_format,
+        $test_cnt, $temp_source, $temp_target, $environment, $no_clean,
+        $SHOW_ONLY_GROUPS, $slow_min_ms, $exts_tested, $end_time, $start_time,
+        $html_output, $html_file, $temp_urlbase, $workers, $workerID;
+
+    $workerID = 0;
+    if (getenv("TEST_PHP_WORKER")) {
+        $workerID = intval(getenv("TEST_PHP_WORKER"), 10);
```

Comment:
You can just use `$workerID = (int) getenv("TEST_PHP_WORKER");`.

Reply:
Oh. Right, yeah. I think I was thinking of JS `parseInt()` when I wrote this.